### ISSUE TITLE
Not an issue, just added DataGroup

### ISSUE THREAD

**Issue by OLaudren** [joel74/POSH-LTM-Rest#118]
Hello,
Here attached a few files I wrote to enable the DataGroup management.
Do you think you could integrate them soon?

Thank you,
[DataGroup.zip](https://github.com/joel74/POSH-LTM-Rest/files/961532/DataGroup.zip)

**Comment by joel74** [comment]
Hi, OLaudren,

Thanks for creating these. I'll check them out shortly. Just fyi, the normal procedure would be to have your own fork of the project, add your changes to the fork and then make a pull request. Then we all could review the changes and merge them into the main branch. For next time. :)

Cheers,
Joel

**Comment by OLaudren** [comment]
Hi Joel,

Oops, sorry...
I found the right way so I will use next time. Sure.
Sorry again.
Thanks,

**Comment by joel74** [comment]
Hi, OLaudren,

In testing, I found that the example given for New-DataGroup didn't work. It's passing in **http** as the type, but the only accepted values for type are **ip** and **string**. Let's follow the UI conventions and accept 'Address' (instead of IP), 'Integer', 'String' and 'ExternalFile' as valid types. Sound good?

Thanks
Joel

**Comment by OLaudren** [comment]
Yes, OK.  I guess ExternalFile would imply a "external" DataGroup so it will require other change. :(

Olivier.

**Comment by joel74** [comment]
From the looks of it, accepting an external file for a data group might not be an easy addition. My recommendation would be to fork the project, make your changes so that New-DataGroup accepts 'Address','String', and 'Integer', put in a note that 'ExternalFile' will be supported in the future, update the example and create a pull request for that.

**Comment by OLaudren** [comment]
I fully agree.
Thanks a lot.

**Comment by joel74** [comment]
Hi, Olivier,
I'm just checking in to see how this is going. 
Cheers,
Joel

**Comment by mtengmo** [comment]
It would be great if you could add support for datagroups.

**Comment by joel74** [comment]
Ok. I'll take a look again at what Olivier submitted, and tweak it a bit. I don't think I'll be adding the External File support at this time, though.

**Comment by mtengmo** [comment]
Thanks, I looked on the code, and it looks like it only exists three functions; Get-Datagroup, New-Datagroup, and Test-Datagroup. 
I was more interested in adding/modify/delete data to/from datagroups. Both String and Address types.

**Comment by joel74** [comment]
@mtengmo , I haven't really used datagroups myself. So what you're interested in is different, I take it, from a Set-DataGroup function, because you're more interested in modifying the actual data, not the properties of the datagroup, is that accurate?

**Comment by mtengmo** [comment]
Datagroup is similar to a table in sql. You can have three datatypes, (integer, ip or string). IP is just a single column with IPs. String is two columns (string, value). Really powerful to use in irule´s. 
I´m interested to add/modify/delete data from already created datagroups.

**Comment by colombeen** [comment]
Will DataGroup support be added anytime soon? Without it we can't use this module in our environment.

**Comment by joel74** [comment]
I will try to find some time to look at this soon so maybe in the next few weeks it will be added.
